core/runtime/impl/wasm_executor.cpp

```diff
@@ -8,23 +8,47 @@
 #include <binaryen/wasm-binary.h>
 #include "runtime/impl/runtime_external_interface.hpp"
 
+OUTCOME_CPP_DEFINE_CATEGORY(kagome::runtime, WasmExecutor::Error, e) {
+  using kagome::runtime::WasmExecutor;
+  switch (e) {
+    case WasmExecutor::Error::EMPTY_STATE_CODE:
+      return "Provided state code is empty, calling a function is impossible";
+    case WasmExecutor::Error::INVALID_STATE_CODE:
+      return "Invalid state code, calling a function is impossible";
+    case WasmExecutor::Error::EXECUTION_ERROR:
+      return "An error occured during call execution";
+  }
+  return "Unknown error";
+}
+
 namespace kagome::runtime {
 
   WasmExecutor::WasmExecutor(std::shared_ptr<extensions::Extension> extension)
       : extension_(std::move(extension)) {}
 
-  wasm::Literal WasmExecutor::call(const common::Buffer &state_code,
-                                   wasm::Name method_name,
-                                   const wasm::LiteralList &args) {
+  outcome::result<wasm::Literal> WasmExecutor::call(
+      const common::Buffer &state_code, wasm::Name method_name,
+      const wasm::LiteralList &args) {
+    if (state_code.size() == 0) {
+      return Error::EMPTY_STATE_CODE;
+    }
+
     wasm::Module module{};
     wasm::WasmBinaryBuilder parser(
         module,
         reinterpret_cast<std::vector<char> const &>(  // NOLINT
             state_code.toVector()),
         false);
-    parser.read();
-
-    return callInModule(module, method_name, args);
+    try {
+      parser.read();
+    } catch (wasm::ParseException& e) {
```

Comment:
One last change please, add printing exception message to log. It has dump(std::ostream&) method. Thanks.